python/ray/util/spark/start_ray_node.py

```diff
@@ -62,7 +62,10 @@
     fcntl.flock(lock_fd, fcntl.LOCK_SH)
     process = subprocess.Popen([ray_cli_cmd, "start", *arg_list], text=True)
     # This makes sure that ray node is started
-    time.sleep(2)
+    for i in range(10):
+        if os.path.exists(os.path.join(temp_dir, "session_latest")):
+            break
+        time.sleep(2)
```

Comment:
assuming user firstly manually run command `ray start --head` (it will create "/local_disk0/tmp/session_latest" folder symlink.
then user run `setup_ray_cluster` (global mode), then the code here will break loop at Line 67 without waiting `session_latest` pointing to new ray cluster.
So I suggest we just do following instead of a loop:
```suggestion
   time.sleep(2)
```

Reply:
time.sleep(2) is not enough, in that case maybe we should wait for at least 10 seconds. From my test it normally takes 5 seconds until the message 'ray head node is started' shows up.

Reply:
Ok.
withdraw https://github.com/ray-project/ray/pull/39313/files#r1324207293

If we have this,
https://github.com/ray-project/ray/pull/39313/files#r1324223649
we can ensure it is a empty ray-temp-dir directory before starting ray node